src/Annotation/ApiSubresource.php

```diff
@@ -20,10 +20,20 @@
  *
  * @Annotation
  * @Target({"METHOD", "PROPERTY"})
+ * @Attributes(
+ *     @Attribute("maxDepth", type="int"),
+ * )
  */
 #[\Attribute(\Attribute::TARGET_PROPERTY | \Attribute::TARGET_METHOD)]
 final class ApiSubresource
 {
+    use AttributesHydratorTrait;
```

Comment:
I am wondering if using this trait is unnecessary/overkill.

A simpler solution would be like @rogierknoester suggested in the issue:

```php
public function __construct($maxDepth = null)
{
    if (!\is_array($maxDepth)) { // @phpstan-ignore-line
        $this->maxDepth = $maxDepth;
        return;
    }
    $this->maxDepth = $maxDepth['maxDepth'];
}
```

Reply:
:+1: maxDepth is the only option you can avoid the trait

Reply:
Perfect, I definitely felt like I was getting into code I had no reason to be in. I'll make the changes.